relay-server/src/services/processor.rs

```diff
@@ -1283,12 +1283,14 @@ impl EnvelopeProcessorService {
         project_id: ProjectId,
         project_info: Arc<ProjectInfo>,
         sampling_decision: SamplingDecision,
-    ) -> Result<(), ProcessingError> {
-        if state.event_metrics_extracted {
-            return Ok(());
+        event_metrics_extracted: EventMetricsExtracted,
+        spans_extracted: SpansExtracted,
+    ) -> Result<EventMetricsExtracted, ProcessingError> {
+        if event_metrics_extracted.0 {
+            return Ok(event_metrics_extracted);
```

Comment:
Decided for functions that take in a value and return a modified version, to directly return the value itself when no modification happens. This is done to simplify the control flow logic on the caller side.